cassiopeia/solver/solver_utilities.py

```diff
@@ -174,22 +187,42 @@ def collapse_unifurcations(tree: ete3.Tree) -> ete3.Tree:
 
 def transform_priors(
     priors: Optional[Dict[int, Dict[int, float]]],
-    prior_function: Optional[Callable[[float], float]],
-):
-    """Generates a dictionary of negative log probabilities from priors.
+    prior_transformation: str = "negative_log",
+) -> Dict[int, Dict[int, float]]:
+    """Generates a dictionary of weights from priors.
 
-    Generates a dictionary of weights for use in algorithms that inherit the
-    GreedySolver from given priors.
+    Generates a dictionary of weights from given priors for each character/state
+    pair for use in algorithms that inherit the GreedySolver. Supported
+    transformations include negative log, negative log square root, and inverse.
 
     Args:
         priors: A dictionary of prior probabilities for each character/state
             pair
-        prior_function: A function defining a transformation on the priors
-            in forming weights
+        prior_transformation: A function defining a transformation on the priors
+            in forming weights. Supports the following transformations:
+                "negative_log": Transforms each probability by the negative log
+                "negative_square_root_log": Transforms each probability by the
+                    square root of the negative log
+                "inverse": Transforms each probability p by taking 1/p
 
     Returns:
         A dictionary of weights for each character/state pair
     """
+    if prior_transformation not in [
+        "negative_log",
+        "negative_square_root_log",
+        "inverse",
+    ]:
+        logging.info("Please select one of the supported prior transformations")
+        raise PriorTransformationError()
+
+    prior_function = lambda x: -np.log(x)
+
+    if prior_transformation == "negative_square_root_log":
+        prior_function = lambda x: -np.log(np.sqrt(x))
```

Comment:
Does this really make sense?

I had thought that it would be sqrt (1/x), and from the docstring I thought it was sqrt(-log(x)). I think when we discussed this earlier this week, it was sqrt(1/x).